### ISSUE TITLE
Error when using column reference as the second argument of the contains function

### ISSUE THREAD

**Issue by guillaumemeyer** [metabase/metabase#31284]
### Describe the bug

It looks like Metabase doesn't accepts column reference as a source for the second argument of the `contains` function.
When creating a custom column with an expression like `case(contains([Display name], [Given name]), "yes", "no")`, the question returns an error `Input to update-string-value does not match schema: [0;33m [(named [(named (not (= :value :field)) :value) nil nil] value) nil] [0m`.

### To Reproduce

Fom a sample `users` table
1. Create a question
2. Add a custom column with an expression such as `case(contains([Display name], [Given name]), "yes", "no")`
3. Show the preview
4. Returns an error `Input to update-string-value does not match schema: [0;33m [(named [(named (not (= :value :field)) :value) nil nil] value) nil] [0m`

### Expected behavior

The expression should accept referenced columns for both contains parameters

### Logs

{:database_id 2,
 :started_at #t "2023-06-02T10:42:44.943921Z[GMT]",
 :error_type :invalid-query,
 :json_query
 {:database 2,
  :query
  {:source-table 15,
   :expressions {:calc ["case" [[["contains" ["field" 1374 nil] ["field" 1372 nil]] "yes"]] {:default "no"}]},
   :fields [["field" 1376 nil] ["field" 1374 nil] ["field" 1372 nil] ["expression" "calc" nil]],
   :limit 10},
  :type "query",
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> driver.sql.query_processor$fn__63286$update_string_value__63293.invoke(query_processor.clj:1070)"
  "driver.sql.query_processor$fn__63317.invokeStatic(query_processor.clj:1080)"
  "driver.sql.query_processor$fn__63317.invoke(query_processor.clj:1078)"
  "driver.sql.query_processor$fn__63080.invokeStatic(query_processor.clj:825)"
  "driver.sql.query_processor$fn__63080.invoke(query_processor.clj:819)"
  "driver.sql.query_processor$fn__62767.invokeStatic(query_processor.clj:522)"
  "driver.sql.query_processor$fn__62767.invoke(query_processor.clj:519)"
  "driver.sql.query_processor$as.invokeStatic(query_processor.clj:984)"
  "driver.sql.query_processor$as.doInvoke(query_processor.clj:953)"
  "driver.sql.query_processor$fn__63260$iter__63262__63266$fn__63267$fn__63268.invoke(query_processor.clj:1052)"
  "driver.sql.query_processor$fn__63260$iter__63262__63266$fn__63267.invoke(query_processor.clj:1051)"
  "driver.sql.query_processor$fn__63260.invokeStatic(query_processor.clj:1051)"
  "driver.sql.query_processor$fn__63260.invoke(query_processor.clj:1049)"
  "driver.sql.query_processor$apply_top_level_clauses$fn__63538.invoke(query_processor.clj:1372)"
  "driver.sql.query_processor$apply_top_level_clauses.invokeStatic(query_processor.clj:1370)"
  "driver.sql.query_processor$apply_top_level_clauses.invoke(query_processor.clj:1366)"
  "driver.sql.query_processor$apply_clauses.invokeStatic(query_processor.clj:1410)"
  "driver.sql.query_processor$apply_clauses.invoke(query_processor.clj:1400)"
  "driver.sql.query_processor$apply_source_query.invokeStatic(query_processor.clj:1394)"
  "driver.sql.query_processor$apply_source_query.invoke(query_processor.clj:1379)"
  "driver.sql.query_processor$apply_clauses.invokeStatic(query_processor.clj:1408)"
  "driver.sql.query_processor$apply_clauses.invoke(query_processor.clj:1400)"
  "driver.sql.query_processor$mbql__GT_honeysql.invokeStatic(query_processor.clj:1433)"
  "driver.sql.query_processor$mbql__GT_honeysql.invoke(query_processor.clj:1424)"
  "driver.sql.query_processor$mbql__GT_native.invokeStatic(query_processor.clj:1442)"
  "driver.sql.query_processor$mbql__GT_native.invoke(query_processor.clj:1438)"
  "driver.sql$fn__101522.invokeStatic(sql.clj:42)"
  "driver.sql$fn__101522.invoke(sql.clj:40)"
  "query_processor.middleware.mbql_to_native$query__GT_native_form.invokeStatic(mbql_to_native.clj:14)"
  "query_processor.middleware.mbql_to_native$query__GT_native_form.invoke(mbql_to_native.clj:9)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__68063.invoke(mbql_to_native.clj:21)"
  "query_processor$fn__70691$combined_post_process__70696$combined_post_process_STAR___70697.invoke(query_processor.clj:243)"
  "query_processor$fn__70691$combined_pre_process__70692$combined_pre_process_STAR___70693.invoke(query_processor.clj:240)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__69083$fn__69088.invoke(resolve_database_and_driver.clj:36)"
  "driver$do_with_driver.invokeStatic(driver.clj:90)"
  "driver$do_with_driver.invoke(driver.clj:86)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__69083.invoke(resolve_database_and_driver.clj:35)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__64953.invoke(fetch_source_query.clj:310)"
  "query_processor.middleware.store$initialize_store$fn__65131$fn__65132.invoke(store.clj:12)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:47)"
  "query_processor.store$do_with_store.invoke(store.clj:41)"
  "query_processor.middleware.store$initialize_store$fn__65131.invoke(store.clj:11)"
  "query_processor.middleware.normalize_query$normalize$fn__69372.invoke(normalize_query.clj:25)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__66309.invoke(constraints.clj:54)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__69308.invoke(process_userland_query.clj:150)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__69685.invoke(catch_exceptions.clj:171)"
  "query_processor.reducible$async_qp$qp_STAR___59455$thunk__59457.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___59455.invoke(reducible.clj:109)"
  "query_processor.reducible$async_qp$qp_STAR___59455.invoke(reducible.clj:94)"
  "query_processor.reducible$sync_qp$qp_STAR___59467.doInvoke(reducible.clj:129)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:362)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:358)"
  "query_processor$fn__70739$process_query_and_save_execution_BANG___70748$fn__70751.invoke(query_processor.clj:373)"
  "query_processor$fn__70739$process_query_and_save_execution_BANG___70748.invoke(query_processor.clj:366)"
  "query_processor$fn__70784$process_query_and_save_with_max_results_constraints_BANG___70793$fn__70796.invoke(query_processor.clj:385)"
  "query_processor$fn__70784$process_query_and_save_with_max_results_constraints_BANG___70793.invoke(query_processor.clj:378)"
  "api.dataset$run_query_async$fn__86545.invoke(dataset.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__54305$fn__54306.invoke(streaming.clj:166)"
  "query_processor.streaming$streaming_response_STAR_$fn__54305.invoke(streaming.clj:165)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:69)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:67)"
  "async.streaming_response$do_f_async$task__36922.invoke(streaming_response.clj:88)"],
 :card_id nil,
 :context :ad-hoc,
 :error
 "Input to update-string-value does not match schema: \n\n\t   [(named [(named (not (= :value :field)) :value) nil nil] value) nil]  \n\n",
 :row_count 0,
 :running_time 0,
 :preprocessed
 {:database 2,
  :query
  {:source-table 15,
   :expressions {"calc" [:case [[[:contains [:field 1374 nil] [:field 1372 nil]] "yes"]] {:default "no"}]},
   :fields [[:field 1376 nil] [:field 1374 nil] [:field 1372 nil] [:expression "calc"]],
   :limit 10},
  :type :query,
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
  :info {:executed-by 2, :context :ad-hoc}},
 :ex-data
 {:type :schema.core/error,
  :value
  [[:field
    1372
    {:metabase.query-processor.util.add-alias-info/source-table 15,
     :metabase.query-processor.util.add-alias-info/source-alias "given_name",
     :metabase.query-processor.util.add-alias-info/desired-alias "given_name",
     :metabase.query-processor.util.add-alias-info/position 2}]
   #object[metabase.driver.sql.query_processor$fn__63317$fn__63321 0x5d978eb "metabase.driver.sql.query_processor$fn__63317$fn__63321@5d978eb"]],
  :error [(named [(named (not (= :value :field)) :value) nil nil] value) nil]},
 :data {:rows [], :cols []}}

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36 Edg/115.0.0.0",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.19+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.19",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.19+7",
    "os.name": "Linux",
    "os.version": "5.10.164.1-1.cm1",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.18"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-04-28",
      "tag": "v0.46.2",
      "branch": "release-x.46.x",
      "hash": "8967c94"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

blocking for non-technical users

### Additional context

I found a very tricky workaround to make it work:
- Change the expression to `case(contains([Display name], "MYVALUE"), "yes", "no")`
- Convert the query to SQL
- From this query, replace `LIKE '%MYVALUE%'` with `LIKE CONCAT('%', "public"."users"."given_name", '%')`.
- Run it, it works!

Basically the trick is to:
- use SQL to be able to reference the `given_name` column properly with `"public"."users"."given_name"`
- surround the `given_name` with `%` so that it performs a search that match if the string is anywhere in the source column

**Comment by guillaumemeyer** [comment]
cc @mnarboni

**Comment by guillaumemeyer** [comment]
Any news?
